src/ar_util.erl

```diff
@@ -40,13 +40,26 @@ pick_random(Xs) ->
 encode(Bin) ->
 	base64url:encode(Bin).
 
-%% @doc Decode a URL safe base64 to binary.
+%% @doc Try to decode a URL safe base64 into a binary or throw an error when
+%% invalid.
 decode([]) -> [];
-decode(Bin) when is_list(Bin) ->
-	decode(list_to_binary(Bin));
+decode(List) when is_list(List) ->
```

Comment:
Tiny naming suggestion: Let's call it `String` instead of `List`.

Reply:
I'd rather not but if you feel strongly about this, can do.